src/orion/storage/track.py

```diff
@@ -0,0 +1,464 @@
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.storage.legacy -- Track Storage Protocol
+====================================================
+
+.. module:: base
+   :platform: Unix
+   :synopsis: Implement a storage protocol to allow Orion to use track as a storage method
+
+"""
+
+from collections import defaultdict
+import datetime
+import logging
+import uuid
+
+try:
+    from track.client import TrackClient
+    from track.persistence.utils import parse_uri
+    from track.serialization import to_json
+    from track.structure import CustomStatus, Status as TrackStatus
+    from track.structure import Project, Trial as TrackTrial, TrialGroup
+
+    HAS_TRACK = True
+except ImportError:
+    HAS_TRACK = False
+
+from orion.core.worker.trial import Trial as OrionTrial
+from orion.storage.base import BaseStorageProtocol
+
+
+log = logging.getLogger(__name__)
+
+
+if HAS_TRACK:
+    _status = [
+        CustomStatus('new', TrackStatus.CreatedGroup.value + 1),
+        CustomStatus('reserved', TrackStatus.CreatedGroup.value + 2),
+
+        CustomStatus('suspended', TrackStatus.FinishedGroup.value + 1),
+        # CustomStatus('completed', TrackStatus.FinishedGroup.value + 2),
+
+        CustomStatus('interrupted', TrackStatus.ErrorGroup.value + 1),
+        CustomStatus('broken', TrackStatus.ErrorGroup.value + 3)
+    ]
+
+    _status_dict = {
+        s.name: s for s in _status
+    }
+    _status_dict['completed'] = TrackStatus.Completed
+
+
+def get_track_status(val):
+    """Convert orion status to track status"""
+    return _status_dict.get(val)
+
+
+def convert_track_status(status):
+    """Convert track status to orion status"""
+    return status.name.lower()
+
+
+def remove_leading_slash(name):
+    """Remove leading slash"""
+    # if name[0] == '/':
+    #     return name[1:]
+    # return name
+    return name
```

Comment:
Not doing it anymore?

Reply:
still planing too but orion complains if I strip them and put them back